src/Commands/core/ArchiveDumpCommands.php

```diff
@@ -0,0 +1,582 @@
+        unset($archive);
+        Phar::unlinkArchive($archivePath);
+        $archivePath .= '.gz';
+
+        if (!$options['destination']) {
+            return $archivePath;
+        }
+
+        if ($this->filesystem->exists($options['destination'])) {
+            if (!$options['overwrite']) {
+                throw new Exception(
+                    'The destination file already exists. Use "--overwrite" option for overwriting an existing file.'
+                );
+            }
+
+            $this->filesystem->remove($options['destination']);
+        }
+
+        $this->logger()->info(
+            dt(
+                'Moving archive file from !from to !to',
+                ['!from' => $archivePath, '!to' => $options['destination']]
+            )
+        );
+        $this->filesystem->rename($archivePath, $options['destination']);
+
+        return $options['destination'];
+    }
+
+    /**
+     * Creates the MANIFEST file.
+     *
+     * @param array $options
+     *   The command options.
+     */
+    private function createManifestFile(array $options): void
+    {
+        $this->logger()->info(dt('Creating !manifest file...', ['!manifest' => self::MANIFEST_FILE_NAME]));
+        $manifest = [
+            'datestamp' => time(),
+            'formatversion' => self::MANIFEST_FORMAT_VERSION,
+            'components' => [
+                self::COMPONENT_CODE => $options['code'],
+                self::COMPONENT_FILES => $options['files'],
+                self::COMPONENT_DATABASE => $options['db'],
+            ],
+            'description' => $options['description'] ?? null,
+            'tags' => $options['tags'] ?? null,
+            'generator' => $options['generator'] ?? 'Drush archive:dump',
+            'generatorversion' => $options['generatorversion'] ?? Drush::getVersion(),
+        ];
+        $manifestFilePath = Path::join($this->archiveDir, self::MANIFEST_FILE_NAME);
+        file_put_contents(
+            $manifestFilePath,
+            Yaml::dump($manifest)
+        );
+    }
+
+    /**
+     * Returns TRUE is the site is a "web" docroot site.
+     *
+     * @return bool
+     *
+     * @throws \Exception
+     */
+    private function isWebRootSite(): bool
+    {
+        return self::WEB_DOCROOT === basename($this->siteAliasManager()->getSelf()->root());
+    }
+
+    /**
+     * Returns site's docroot name.
+     *
+     * @return string
+     *
+     * @throws \Exception
+     */
+    private function getDocrootName(): string
+    {
+        return $this->isWebRootSite() ? self::WEB_DOCROOT : '';
+    }
+
+    /**
+     * Creates "code" archive component and returns the absolute path.
+     *
+     * @param array $options
+     *  The command options.
+     *
+     * @return string
+     *  The full path to the code archive component directory.
+     *
+     * @throws \Exception
+     */
+    private function getCodeComponentPath(array $options): string
+    {
+        $codePath = $this->isWebRootSite()
```

Comment:
Once you switch to the bootstrap manager, the method `getComposerRoot()` will return the project root (code path)

Reply:
Updated